src/walletd/runtime.rs

```diff
@@ -360,8 +360,18 @@ impl Runtime {
                 error!("Msg and source don't have same swap_id, ignoring...");
                 return Ok(());
             }
+
+            Request::Protocol(msg)
```

Comment:
Why are you adding these permission checks? I don't think we should firewall the services that are on a common service bus from each other.

Reply:
> I disagree, there is no reason to leave an attack surface open for exploiting all the bugs we have left

This "permission" check does not add any actual security in my eyes. Our internal messages are unauthenticated and can be read out, altered and re-routed as an attacker pleases. We already have a firewall in place in peerd that only allows whitelisted messages into the internal message busses and only passes protocol messages to hardcoded services. Any additional internal checks are easily evaded.

That said, if this increases the code robustness, that is a fine reason for me.